storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -4828,6 +4828,8 @@ void fsp_init_resume_alter_encrypt_tablespace() {
   THD *thd = create_thd(false, true, true, 0);
 #endif
 
+  thd->set_new_thread_id();
+
```

Comment:
Can we add this as part of create_thd() API? I think it is reasonable to add sensible defaults after THD creation? Is query_id assigned or skipped?

Reply:
At the moment my understanding of THD lifetime is very fragmented. So I don't think I can provide an answer at the moment. Need to investigate further.

There are a lot of places which allocate instances of `THD`, it's not limited `create_thd()`. So I'd expect `m_thread_id` initialization to go to the constructor, `THD::THD`. In fact, it's initialized to `Global_THD_manager::reserved_thread_id` there, which is constant `0`. I assumed that since the actual assignment of the identifier is separated from the constructor, there could be reasons.

As for `query_id`, this thread and its action are initiated from an internal intent. It's not associated with a client's query. It probably won't hurt, but I need to look into code more to figure that out.

Reply:
Looks like list of threads is maintained in `Global_THD_manager` in two separate places. One in a simple array `thread_ids`, which is not used for anything other than keeping an ID from a thread. Another is partitioned array `thd_list` which is used for example to list processes in `SHOW PROCESSLIST`. So just adding a thread ID for every thread should not have any visible effects. There may be issues with short lived threads if `THD::release_resources()` is not called before their destruction. But that will trigger an assertion in `Global_THD_manager::~Global_THD_manager()`, so the change should be relatively safe.

Assignment of `query_id` shouldn't matter since this thread is never made visible anyway.